### ISSUE TITLE
The axis of softmax supposed to be 0?

### ISSUE THREAD

**Issue by KylinA1** [anonymous0908X/KHGT#6]
```python
def GAT(self, srcEmbeds, tgtEmbeds, tgtNodes, maxNum, Qs, Ks, Vs):
	QWeight = tf.nn.softmax(NNs.defineRandomNameParam([args.memosize, 1, 1], reg=True), axis=1)
	KWeight = tf.nn.softmax(NNs.defineRandomNameParam([args.memosize, 1, 1], reg=True), axis=1)
	VWeight = tf.nn.softmax(NNs.defineRandomNameParam([args.memosize, 1, 1], reg=True), axis=1)
	Q = tf.reduce_sum(Qs * QWeight, axis=0)
	K = tf.reduce_sum(Ks * KWeight, axis=0)
	V = tf.reduce_sum(Vs * VWeight, axis=0)
```
 I found above code to implement equation (3), I have 2 questions:
1. I think it should be below:
```python
tf.nn.softmax(..., axis = 0) 
```
2. For all behavior $k$, there supposed to be only 1 group of $Q^h_m, m = 1\cdots, M$ to aggregated from, according to equation (3). While in practical implementation, each behavior $k$ has its own $M$ channels basis parameters.

Please excuse me if there is any misunderstanding.

**Comment by akaxlh** [comment]
Thank you for your interests in our work. For your questions:
1. Yes, indeed the softmax seems not working actually. Though the softmax normalization is not obligated. We can still regard this as a non-normalized weighting.
2. Actually the Q K V mappings are shared among different behavior-specific interaction data, which is stored in the varaiable mats. We don't share Q K V across graph layers and between user/item modeling.
Again, thank you for your questions!
